controllers/mutation_reconcile_looper.go

```diff
@@ -571,3 +495,104 @@ func getStructFieldValue(v *cue.Struct, field string) (string, error) {
 	}
 	return f.Value.String()
 }
+
+func (m *MutationReconcileLooper) getSource(ctx context.Context, ref v1alpha1.PatchStrategicMergeSourceRef) (sourcev1.Source, error) {
+	var sourceObj sourcev1.Source
+	key := types.NamespacedName{
+		Name:      ref.Name,
+		Namespace: ref.Namespace,
+	}
+	switch ref.Kind {
+	case sourcev1.GitRepositoryKind:
+		var repository sourcev1.GitRepository
+		err := m.Client.Get(ctx, key, &repository)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return sourceObj, err
+			}
+			return sourceObj, fmt.Errorf("unable to get source '%s': %w", key, err)
+		}
+		sourceObj = &repository
+	case sourcev1.BucketKind:
+		var bucket sourcev1.Bucket
+		err := m.Client.Get(ctx, key, &bucket)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return sourceObj, err
+			}
+			return sourceObj, fmt.Errorf("unable to get source '%s': %w", key, err)
+		}
+		sourceObj = &bucket
+	case sourcev1.OCIRepositoryKind:
+		var ocirepo sourcev1.OCIRepository
+		err := m.Client.Get(ctx, key, &ocirepo)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return sourceObj, err
+			}
+			return sourceObj, fmt.Errorf("unable to get source '%s': %w", key, err)
+		}
+		sourceObj = &ocirepo
+	default:
+		return sourceObj, fmt.Errorf("source `%s` kind '%s' not supported", ref.Name, ref.Kind)
+	}
+
+	return sourceObj, nil
+}
+
+func (m *MutationReconcileLooper) fetchArtifact(ctx context.Context, artifact *sourcev1.Artifact) (*bytes.Buffer, error) {
```

Comment:
This is actually unused and the http client setup only happens in the configuration controller. This would probably fail with a nil error in case of the localization where the setup doesn't happen. 🤔 Is this perhaps just a left in?

Reply:
Thanks for catching, yes left in. We use the flux fetcher package now instead. Will remove.